core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/ThrottlingAllocationDecider.java

```diff
@@ -102,17 +108,16 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
 
     @Override
     public Decision canAllocate(RoutingNode node, RoutingAllocation allocation) {
-        int currentRecoveries = 0;
-        for (ShardRouting shard : node) {
-            if (shard.initializing()) {
-                currentRecoveries++;
-            }
-        }
-        if (currentRecoveries >= concurrentRecoveries) {
+        int currentOutRecoveries = allocation.routingNodes().getOutgoingRecoveries(node.nodeId());
+        int currentInRecoveries = allocation.routingNodes().getIncomingRecoveries(node.nodeId());
+        if (currentOutRecoveries >= concurrentOutgoingRecoveries) {
+            return allocation.decision(Decision.THROTTLE, NAME, "too many shards currently recovering [%d], limit: [%d]",
```

Comment:
Can you add the word "outgoing" to the description so we can differentiate it with the incoming message?